backend/backend-python/handler.py

```
@@ -1,45 +1,761 @@
 
-    def upload_handler(self, reqs):
+    def embed_image(self, reqs: list[message.EmbedImageRequest]):
+        """
+        Embeds images into the specified embed pointers.
+        """
+        for req in reqs:
+            if len(req.embed_ptrs) > self.max_num_embeds:
+                raise ValueError(
+                    f"Number of embed pointers {len(req.embed_ptrs)} exceeds "
+                    f"maximum {self.max_num_embeds}."
+                )
```

Ruff (0.13.2)

Avoid specifying long messages outside the exception class

(TRY003)